File: pvtapp/src/main/java/com/freshm/pvtapp/service/ProductService.java
package com.freshm.pvtapp.service;


import com.freshm.pvtapp.dto.ProductRequest;
import com.freshm.pvtapp.dto.ProductResponse;

import java.util.List;


public interface ProductService {


    /**
     * Create new product
     */
    ProductResponse createProduct(
            ProductRequest request
    );



    /**
     * Get all products of logged-in company
     */
    List<ProductResponse> getAllProducts();



    /**
     * Get single product
     */
    ProductResponse getProductById(
            Long productId
    );



    /**
     * Update existing product
     */
    ProductResponse updateProduct(
            Long productId,
            ProductRequest request
    );



    /**
     * Disable product
     */
    void deleteProduct(
            Long productId
    );


}

File: freshm-frontend/src/services/productService.js
import api from "./api";

// Create Product (Article)
export const createProduct = async (productData) => {
  const response = await api.post("/products", productData);
  return response.data;
};

// Get All Products
export const getAllProducts = async () => {
  const response = await api.get("/products");
  return response.data;
};

// Get Product By Id
export const getProductById = async (id) => {
  const response = await api.get(`/products/${id}`);
  return response.data;
};

// Update Product
export const updateProduct = async (id, productData) => {
  const response = await api.put(`/products/${id}`, productData);
  return response.data;
};

// Delete Product
export const deleteProduct = async (id) => {
  const response = await api.delete(`/products/${id}`);
  return response.data;
};